internal/storagenode/admin_server.go

```diff
@@ -60,7 +124,19 @@ func (as *adminServer) Sync(ctx context.Context, req *snpb.SyncRequest) (*snpb.S
 			LogStreamID: req.LogStreamID,
 		},
 	})
-	return &snpb.SyncResponse{Status: syncStatus}, err
+	if err != nil {
+		var code codes.Code
+		switch err {
```

Comment:
How about integrating with the L103 code?

Reply:
Are you suggesting refactoring, such as extracting the common switch block?

Reply:
If you mean extracting out and integrating the common switch block, I want instead to propose another approach, such as intercepting common errors like ErrClosed and ErrNotExist.

Reply:
I mean to create a separate function that replaces snerrors with grpc error code. There is no need to do that if 1:1 matching is not possible.